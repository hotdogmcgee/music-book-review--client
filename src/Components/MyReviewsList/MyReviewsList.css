.my-reviews-item-info {
    display: flex;
    flex-direction: column;
}