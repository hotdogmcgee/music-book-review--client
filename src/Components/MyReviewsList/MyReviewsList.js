import React from "react";
import { Link } from "react-router-dom";
import { Section, NiceDate } from "../Utils/Utils";
import "./MyReviewsList.css";

export default class MyReviewsList extends React.Component {
  //to-do:  make a path/modal for updating a review

  renderMyReviews() {
    const reviews = this.props.reviews;
    const bookList = this.props.bookList;

    if (reviews) {
      const reviewsItems = reviews.map((rv, index) => {
        let book = bookList.filter(book => rv.book_id === book.id);

        rv.title = book[0] ? book[0].title : "loading";
        rv.authors = book[0] ? book[0].authors : "";
        // rv.authors = book[0].authors

        return (
          <div className="my-reviews-item" key={index}>
            <div className="my-reviews-item-book-info">
              <Link to={`/book/${rv.book_id}`}>{rv.title}</Link>
              <div className="authors-container">
                {rv.authors ? (
                  <RenderAuthors authors={rv.authors} />
                ) : (
                  "no authors!"
                )}
              </div>
            </div>

            <div className="my-reviews-item-data">
              <p>{rv.rating}</p>

              <p>{rv.review_text}</p>
              <p>
                <NiceDate date={rv.date_created} />
              </p>
            </div>
          </div>
        );
      });

      return <ul id="my-reviews-list">{reviewsItems}</ul>;
    } else {
      return <p>No reviews yet!</p>;
    }
  }

  render() {
    return (
      <Section className="my-reviews-container">
        {this.renderMyReviews()}
      </Section>
    );
  }
}

function RenderAuthors({ authors }) {

  return authors.map((author, key) => {
    const commaSpace = key < authors.length - 1 ? ", " : "";

    return (
      <span className="author-container" key={key}>
        {`${author.first_name} ${author.last_name}${commaSpace} `}
      </span>
    );
  });
}
