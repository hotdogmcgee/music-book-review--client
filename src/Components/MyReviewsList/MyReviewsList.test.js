import React from "react";
import MyReviewsList from "./MyReviewsList";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("MyReviewsList component", () => {
  it("renders a MyReviewsList by default", () => {
    const wrapper = shallow(<MyReviewsList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
