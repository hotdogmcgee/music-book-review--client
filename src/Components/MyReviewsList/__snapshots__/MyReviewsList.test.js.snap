// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReviewsList component renders a MyReviewsList by default 1`] = `
<Section
  className="my-reviews-container"
>
  <p>
    No reviews yet!
  </p>
</Section>
`;
